src/web/mjs/connectors/Cycomi.mjs

```diff
@@ -8,54 +8,133 @@ export default class CyComi extends Connector {
-                title: element.querySelector('.card-texts-title').textContent.trim()
-            };
-        });
+    async _getChapters(manga) {
+        let cursor = undefined;
+        const chapterList = [];
+        do {
+            const [chapters, nextCursor] = await this._getChaptersFromCursor(manga, cursor);
+            chapterList.push(...chapters);
+            cursor = nextCursor;
+        } while (cursor);
+        return chapterList;
     }
 
-    async _getChapters(manga) {
-        const uri = new URL(manga.id, this.url);
+    async _getChaptersFromCursor(manga, cursor) {
+        const params = new URLSearchParams({
+            limit: 20,
+            titleId: manga.id,
+            sort: 2,
+        });
+        if (cursor) {
+            params.append("cursor", cursor);
+        }
+        const uri = new URL("/api/chapter/paginatedList?" + params.toString(), this.url);
         const request = new Request(uri, this.requestOptions);
-        const data = await this.fetchDOM(request, '.title-chapters > a');
-        return data.map(element => {
+        const response = await this.fetchJSON(request);
+        const chapters = response.data.map(element => {
             return {
-                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
-                title: element.querySelector('.chapter-title').textContent.trim(),
+                id: {"chapterId": element.id, "titleId": manga.id},
+                title: element.name + " " + element.subName,
                 language: '',
             };
-        }).reverse();
+        });
+        return [chapters, response.nextCursor];
+    }
+
+    s(e, t) {
+        let n = (e=>{
+                let t = new Uint8Array(256);
+                t.forEach((e, n) =>{
+                    t[n] = n;
+                });
+                let n = 0;
+                return t.forEach((i, r) =>{
+                    n = (n + t[r] + e.charCodeAt(r % e.length)) % 256;
+                    let l = t[r];
+                    t[r] = t[n],
+                    t[n] = l;
+                }),
+                t;
+            }) (t),
+            i = 0,
+            r = 0,
+            l = new Uint8Array(e.length);
+        for (let t = 0, a = e.length; t < a; t++) {
+            r = (r + n[i = (i + 1) % 256]) % 256;
+            let a = n[i % 256];
+            n[i % 256] = n[r],
+            n[r] = a;
+            let o = n[(n[i] + n[r]) % 256];
+            l[t] = o ^ e[t];
+        }
+        return l;
+    }
+
+    d(e) {
+        return new Promise((t, n) =>{
+            let i = new FileReader;
+            i.addEventListener('error', n),
+            i.addEventListener('load', () =>t(i.result)),
+            i.readAsDataURL(new Blob([e]));
+        });
     }
 
     async _getPages(chapter) {
-        const uri = new URL(chapter.id, this.url);
+        const uri = new URL("/api/chapter/page/list", this.url);
```

Comment:
_getPages is only supposed to return images url array. 

If you have to perform additional steps (like decrypting / descrambling) you need to use return an array of `this.createConnectorURI(pictureurl) ` and implement _handleConnectorURI and perfom the actual decryption inside.

Please look for samples in connectors code.

Also, filtering can be done in _getPages directly since you are filtering on page URI.
Something like that : 

```
        let pages = response.data.pages.filter(page => page.type == "image");
        pages = pages.filter(page=> page.test(/\/([0-9a-zA-Z]{32})\//));
        return pages.map(page => this.createConnectorURI(page));
}

async _handleConnectorURI(payload) {
        const response = await fetch(request);
        const data = await response.arrayBuffer();
        const enc = new Uint8Array(data);
        const key = uri.match(/\/([0-9a-zA-Z]{32})\//)[1];
   ....     
}
```

You have then to return a buffer with proper myple type so decode the base64 yourself.

```
                data = {
                    mimeType: response.headers.get('content-type'),
                    data: //arry of byte 
                };
                this._applyRealMime(data);
                return data;

 ```